oci/private/tarball.bzl

```diff
@@ -130,7 +175,8 @@ def _tarball_impl(ctx):
         runfiles.append(ctx.file.loader)
 
     return [
-        DefaultInfo(files = depset([tarball]), runfiles = ctx.runfiles(files = runfiles), executable = exe),
+        DefaultInfo(files = depset([mtree_spec]), runfiles = ctx.runfiles(files = runfiles), executable = exe),
```

Comment:
`runfiles` here does point to the output of the `tar` action. So I'm not sure this PR actually does what we want, unless the user builds with `--nobuild_runfile_links` ?

Reply:
confirmed that even with that flag, the tar is still created because `runfiles` are included in the semantics of "default outputs"

Reply:
I can't really imagine a world where by default you would want a built docker tar in runfiles where it would be a lot of hassle to grab the output group... Docker tar + runfiles feels quite niche.

Reply:
Sahin suggested that we just run `tar` at runtime when you execute the loader script, that's the correct answer because more sophisticated per-layer loaders don't need a tar input at all.